app/forms/keywords_form.rb

```diff
@@ -0,0 +1,47 @@
+# frozen_string_literal: true
+
+require 'csv'
+
+class KeywordsForm
+  include ActiveModel::Model
+
+  validates_with KeywordsFormValidator
+
+  attr_accessor :file, :user, :keywords
+
+  def save(params)
+    assign_attributes(params)
+
+    return false if invalid?
+
+    begin
+      if parse_keywords(params[:file])
+        Keyword.create(parse_keywords_from_file(keywords)).map { |keyword| keyword['id'] }
```

Comment:
@aetheryna As you will create a list of keywords and handle, should we use [transaction](https://medium.com/@kristenrogers.kr75/rails-transactions-the-complete-guide-7b5c00c604fc) here?

Reply:
Agreed changed it to a Transaction block 😎 Updated here https://github.com/aetheryna/google-scraper-ruby/pull/34/commits/40f76b1ebffdddf0f3a49ba6f8c7d370bf2afcad